### ISSUE TITLE
Add presentation attributes to svg elements to allow setting after creation and autocomplete

### ISSUE THREAD

**Issue by wholden** [cduck/drawSvg#29]
Hi @cduck , really enjoying the package you put together. From the title of my post, there were two things I thought would be natural to be able to do with this library: (1) set attributes (e.g. width, stroke) after creation of an object and (2) have available attributes exposed in a way that they can easily be seen with autocomplete, for example in a Jupyter notebook or ipython. I didn't immediately see these capabilites in the library, but if they are there and I missed it, please let me know.

At any rate, I took a swing at the problem. (1) was relatively straightforward, but (2) turned out to be a challenging (albeit fun) problem to solve.

I made a demo showing the functionality for the Circle svg element. (https://github.com/wholden/drawSvg/tree/with-presentation-attritbutes) I added an attribute "presentation" which contains the "most notable" editable svg attributes

Basically, I created the ability to dynamically add property attributes to the class, then populated it based on the list of "most notable" presentation attributes from the svg specification (https://developer.mozilla.org/en-US/docs/Web/SVG/Element/circle#Attributes).

Let me know what you think of this idea.

Here is a demo of the result:

![drawsvg_presentation_attributes_demo](https://user-images.githubusercontent.com/17480769/97069579-1bbb6900-1586-11eb-920f-51f9617a6365.gif)

**Comment by cduck** [comment]
Thanks.  (1) Currently `circle.args['fill']` works but could be nicer.

(2) It would be great to support autocomplete as `shape.attr_name` and/or `shape.args/attr/presentation.attr_name` but also allow unknown attribute names to still work.  Your demo solution is fancy but using `__dir__` and `__setattr__` might be simpler and more efficient than creating a property for every possible attribute and also allow unknown SVG attributes.

This could all be implemented in the element parent class where each concrete class only needs to specify its set of supported presentation attributes as a class attributes.

Also, what are your thoughts for allowing `shape.attr_name` directly for convenience?

**Comment by wholden** [comment]
Thanks for the quick reply. I realized (after posting of course), that I had used `circle.args['fill']` implicitly in my solution and therefore answered my own (1) question `forehead-slap*`.

I like your idea of implementing `__setattr__`. There could just be a couple lines of code in `__setattr__` that checks the attribute against a list of valid presentation attribute, and writes it into `self.args[attr]` if it is valid. I think that would knock out two birds with one stone because then it would handle unknown SVG attributes (which would not be in the valid presentation attribute list). That's probably simpler than the property approach, which you may have noticed required some stackoverflow help to figure out how to dynamically assign properties to an instance instead of a class.

My gut instinct is no not do `shape.attr_name` because I liked how `shape.attr` was not too cluttered. Especially for the Path element, it was easy to scroll through autocomplete to see valid path commands (M, A, L, etc.). Doing `shape.args.attr` makes coding sense in the context of how the code is currently written, but when I was exploring the package in Jupyter, it wouldn't have occurred to me to peek inside args to change appearance. I think the `shape.presentation.attr_name` attribute is not descriptive enough perhaps but just based it on the SVG docs.  Maybe `shape.appearance.attr_name` would be better (then it would also be near the top of autocomplete).

**Comment by wholden** [comment]
That being said, the simplicity of `shape.attr_name` is hard to beat. Maybe I wouldn't want to populate it with ALL possible SVG presentation attributes. Maybe just the most common ones (fill, fill_opacity, stroke, strok_width, etc.), then dump the less common ones somewhere more hidden, maybe `shape.svgattributes.attr_name` or something.

**Comment by cduck** [comment]
Closing as stale.  Please re-open if you want to revisit this.
